packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/AppTests.kt

```diff
@@ -124,6 +124,10 @@ class AppTests {
                     // There doesn't seem to be a reliable way to throw "InvalidCredentials" for these.
                     null
                 }
+                AuthenticationProvider.CUSTOM_FUNCTION -> {
+                    // There is no way to capture custom authentication credentials exceptions.
```

Comment:
I'm not sure this is true? What happens if the remote auth function fails or throws an exception?

Reply:
There might be other reasons to throw than `InvalidCredentials`, the function might fail while communicating with an external service for example.

Functions are user-defined and there is no standard way to communicate that there was an exception in the credentials. All exceptions in the remote function are communicated as AuthException with some standard message appending the user error.

Reply:
Good point. I guess that is also fair. I can see that we also document in `App.login` that it will throw an AuthException. I don't see any tests verifying that we are getting an AuthException though?

Reply:
Oh I was fiddling with the behaviors and forgot to add the tests.